File: app/domain/context.py
from dataclasses import dataclass, field


@dataclass
class OrderContext:
    location_id: str
    tenant_id: str
    security_zone_id: str
    network_id: str | None = None


@dataclass
class ResolvedContext:
    location: dict
    tenant: dict
    security_zone: dict
    network: dict | None
    available_networks: list[dict] = field(default_factory=list)

File: frontend/src/types/context.ts
export interface Location {
  id: string
  name: string
  code: string
  region: string
}

export interface Tenant {
  id: string
  name: string
  code: string
}

export interface SecurityZone {
  id: string
  name: string
  level: number
  description: string
}

export interface Network {
  id: string
  name: string
  cidr: string
  type: string
  location_id: string
  security_zone_id: string
}

export interface OrderContext {
  location_id: string
  tenant_id: string
  security_zone_id: string
  network_id?: string
}

export interface ResolvedContext {
  location: Location
  tenant: Tenant
  security_zone: SecurityZone
  network: Network | null
  available_networks: Network[]
}
